ironmon_tracker/Program.lua

```diff
@@ -152,11 +158,22 @@ function Program.updatePokemonTeams()
 
 			if Program.validPokemonData(newPokemonData) then
 				-- Sets the player's trainerID as soon as they get their first Pokemon
-				if Tracker.Data.trainerID == nil or Tracker.Data.trainerID == 0 then
-					Tracker.Data.trainerID = newPokemonData.trainerID
+				if Tracker.Data.isNewGame and (newPokemonData.trainerID ~= nil or newPokemonData.trainerID ~= 0) then
```

Comment:
The `(newPokemonData.trainerID ~= nil or newPokemonData.trainerID ~= 0)` part seems unneeded. Mostly because we just passed the `validPokemonData()` check, so I'd expect this to be a real Pokemon; real data means a real TrainerID.

Either way, the OR condition kind of fails here to serve anypurpose, because if `trainerID` is `0` for example, it will pass the first half of the check, and if its `nil` it will pass the second half.